app/api/endpoints.ts

```diff
@@ -25,3 +26,16 @@ export const GraphQLEndpoints: GraphQLEndpointsMap = {
   [ChainId.SEPOLIA]: requireEnv('NEXT_PUBLIC_SEPOLIA_GRAPH_URL', process.env.NEXT_PUBLIC_SEPOLIA_GRAPH_URL || 'http://localhost:4000/'),
   [ChainId.UNKNOWN]: undefined,
 }
+
+export function getVercelHostPrefix() {
+  if (!isOnServer())
+    return '' // will use relative path on client
+
+  if (!VERCEL_ENV)
+    return 'http://localhost:3000' // local
+
+  if (VERCEL_ENV === 'production')
+    return 'https://eonian.finance' // production
+
+  return process.env.VERCEL_URL ?? '' // preview
```

Comment:
Can we use for API requests only relative URLs? Because this vercel host is much more complicated in reality.

I have high suspicion that it should be more like that
```suggestion
  if (VERCEL_ENV === 'production')
    return VERCEL_PROJECT_PRODUCTION_URL || NEXT_PUBLIC_APP_URL 
 || 'https://eonian.finance' // production

  return VERCEL_BRANCH_URL || VERCEL_URL ?? '' // preview
```

We had few issues with mirrors for metadata rendering for different vercel env that hard to spot directly, so I made investigation and full host building should be more like this https://github.com/eonian-core/web/blob/main/app/layout.tsx#L66. 
In some cases VERCEL_URL is not correct for some previews and `eonian.finance` will work in general for production, but we have `eonian.io` that I trying to keep able to override in emergency case through `NEXT_PUBLIC_APP_URL`, but it also good to keep working in parallel without override through `VERCEL_PROJECT_PRODUCTION_URL`.

So... please check is it possible to keep all request URLs relative, if not than please check this variables description in doc. Maybe I wrote accurate solution for metadata, but it not fully accurate for API

Reply:
done